pynapple/io/interface_npz.py

```diff
@@ -6,18 +6,78 @@
 # @Last Modified by:   Guillaume Viejo
 # @Last Modified time: 2024-04-02 14:32:25
 
-"""
-File classes help to validate and load pynapple objects or NWB files.
-Data are always lazy-loaded.
-Both classes behaves like dictionary.
-"""
 
 import os
 
 import numpy as np
 
 from .. import core as nap
 
+#
+EXPECTED_ENTRIES = {
+    "TsGroup": {"t", "start", "end", "index"},
+    "TsdFrame": {"t", "d", "start", "end", "columns"},
+    "TsdTensor": {"t", "d", "start", "end"},
+    "Tsd": {"t", "d", "start", "end"},
+    "Ts": {"t", "start", "end"},
+    "IntervalSet": {"start", "end"},
+}
+
+
+def _find_class_from_variables(file_variables, data_ndims=None):
+    if data_ndims is not None:
+        # either TsdTensor or Tsd:
+        assert EXPECTED_ENTRIES["Tsd"].issubset(file_variables)
+
+        return "Tsd" if data_ndims == 1 else "TsdTensor"
+
+    for possible_type, espected_variables in EXPECTED_ENTRIES.items():
+        if espected_variables.issubset(file_variables):
+            return possible_type
+
+    return "npz"
+
+
+class LazyNPZLoader:
+    """Class that lazily loads an NPZ file."""
+
+    def __init__(self, file_path, lazy_loading=False):
+        self.lazy_loading = lazy_loading
+        self.file_path = file_path
+        self.npz_file = np.load(
+            file_path, allow_pickle=True, mmap_mode="r" if lazy_loading else None
+        )
+        self.data = {key: None for key in self.npz_file.keys()}
```

Comment:
is an ".npz" memmappable? i thought this was not supported

Reply:
True, it is not, the option there was a leftover. What actually implements the lazy loading is the whole new dedicated class

Reply:
But yeah, I was fooling myself. There seems to be absolutely no way to load lazily from a `npz` file, I tried many options and none seems to work. If you do not have more suggestions I'll drop this for now.